### PR TITLE
feature/ie-36 force delete user

### PR DESCRIPTION
**Pull request by Olivex727**
Draft PR so that code can be reviewed while unit tests are being prepared

### CODE REVIEW

File: src/scripts/delete-user.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import * as O from 'fp-ts/lib/Option';
 import { flow, pipe } from 'fp-ts/lib/function';
 import * as Console from 'fp-ts/lib/Console';
 
-const deleteUser = (
+export const deleteUser = (
```
Comment (Olivex727) [inline src/scripts/delete-user.ts:12]: > Write doc now that this is exported.

What do you mean by this comment?

File: src/scripts/__fixtures__/data-providers.ts [hunk 2]
```diff
@@ -17,6 +17,15 @@ export const user: User = {
   deactivated: false,
 };
 
+export const deactivatedUser: User = {
```
Comment (Olivex727) [inline src/scripts/__fixtures__/data-providers.ts:20]: > Better to use the spread operator to re-use bits of base user.

How does the spread operator work? Which module is it from?

**Comment by Olivex727** [comment]
@dspasojevic ready for a re-review

File: src/scripts/__fixtures__/data-providers.ts (hunk 2 above)
Comment (Olivex727) [reply src/scripts/__fixtures__/data-providers.ts:20]: Changes will be implemented. Didn't know the spread operator was the '...' operator

File: src/scripts/__fixtures__/data-providers.ts [hunk 3]
```diff
@@ -17,6 +17,15 @@ export const user: User = {
...
+  id: 'deactivated_user_id',
```
Comment (dspasojevic) [inline src/scripts/__fixtures__/data-providers.ts:21]: Still not using spread operator.

File: src/scripts/delete-user.test.ts [hunk 4]
```diff
@@ -0,0 +1,179 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+/* eslint-disable functional/no-return-void */
+/* eslint-disable functional/no-expression-statement */
+/* eslint-disable functional/functional-parameters */
+import * as TE from 'fp-ts/TaskEither';
+import * as O from 'fp-ts/Option';
+import { UserService } from './services/user-service';
+import { deleteUser } from './delete-user';
+
+import * as test from './__fixtures__/data-providers';
```
Comment (dspasojevic) [inline src/scripts/delete-user.test.ts:10]: Import baseUserService, etc explicitly.

File: src/scripts/__fixtures__/data-providers.ts (hunk 3 above)
Comment (Olivex727) [reply src/scripts/__fixtures__/data-providers.ts]: Was changed in the most recent commit

File: src/scripts/delete-user.test.ts (hunk 4 above)
Comment (Olivex727) [reply src/scripts/delete-user.test.ts:10]: Should I implement the change for just delete-user.ts?

File: src/scripts/delete-user.ts (hunk 1 above)
Comment (dspasojevic) [reply src/scripts/delete-user.ts:12]: exported (aka public) members should have typedoc.

File: src/scripts/delete-user.test.ts (hunk 4 above)
Comment (dspasojevic) [reply src/scripts/delete-user.test.ts:10]: Just this file. Keep PR small.

Comment (Olivex727) [reply src/scripts/delete-user.test.ts:10]: Will do

File: src/scripts/delete-user.ts (hunk 1 above)
Comment (Olivex727) [reply src/scripts/delete-user.ts:12]: So you mean have some sort of d.ts document. Where should I put it (i.e. which directory), what is an appropriate name to call it, and should it include all of the exports across the scripts section of the oktagon project?

Comment (Olivex727) [reply src/scripts/delete-user.ts:19]: Typedocs added

**Comment by Olivex727** [comment]
@dspasojevic ready for another re-review

File: src/scripts/delete-user.test.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+      ...baseUserService(),
+      deleteUser: () => TE.right(deactivatedUser),
+      getUser: () => TE.right(O.some(deactivatedUser)),
+    };
+
+    // When we attempt to delete an existing deprovisioned user
+    const result = await deleteUser(userService, deactivatedUser.id, false)();
+
+    // Then we should have a right
+    expect(result).toEqualRight(deactivatedUser);
```
Comment (dspasojevic) [inline src/scripts/delete-user.test.ts:29]: One last change; assert that `deleteUser` was called here.

Comment (Olivex727) [reply src/scripts/delete-user.test.ts:29]: Asserted

**Pull request merged**